api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,117 @@
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  private DbWorkspace lookupWorkspace(String workspaceNamespace)
+      throws org.pmiops.workbench.exceptions.NotFoundException {
+    return workspaceDao
+        .getByNamespace(workspaceNamespace)
+        .orElseThrow(() -> new NotFoundException("Workspace not found: " + workspaceNamespace));
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String googleProject = dbWorkspace.getGoogleProject();
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false);
+      responseList.sort(
+          new Comparator<LeonardoListPersistentDiskResponse>() {
+            @Override
+            public int compare(
+                LeonardoListPersistentDiskResponse t1, LeonardoListPersistentDiskResponse t2) {
+              return -1 * t1.getId().compareTo(t2.getId());
+            }
+          });
+      response =
+          !responseList.isEmpty()
+              ? leonardoNotebooksClient.getPersistentDisk(
+                  googleProject, responseList.get(0).getName())
+              : null;
+
+      if (response != null && LeonardoRuntimeStatus.ERROR.equals(response.getStatus())) {
+        log.warning(
+            String.format(
+                "Observed Leonardo runtime with unexpected error status:\n%s",
+                response.getStatus()));
+      }
+
+      if (response != null) {
+        return ResponseEntity.ok(leonardoMapper.toApiDisk(response));
+
+      } else {
+        return ResponseEntity.ok(null);
+      }
+
+    } catch (NotFoundException e) {
```

Comment:
no need to catch. NotFoundException  will be automattly converted to http 404

Reply:
not addressed